pkg/util/system/common_linux.go

```diff
@@ -31,11 +31,54 @@ import (
 	"regexp"
 	"strconv"
 	"strings"
+	"sync"
+	"syscall"
+	"time"
 	"unicode"
 
+	"github.com/cakturk/go-netstat/netstat"
 	"k8s.io/klog/v2"
 )
 
+var (
+	portToPidMap = sync.Map{}
+)
+
+// PortToPid Query pid by tcp port number with the help of go-netstat
+// note: Due to the low efficiency of full traversal, we use a map to cache the results and verify each time
+func PortToPid(port int) (int, error) {
+	if port < 0 {
+		return -1, fmt.Errorf("failed to get pid, the port is invalid")
+	}
+	if pid, ok := portToPidMap.Load(port); ok {
+		if err := syscall.Kill(pid.(int), 0); err == nil {
+			return pid.(int), nil
+		}
+		portToPidMap.Delete(port)
+	}
+	socks, err := netstat.TCPSocks(func(entry *netstat.SockTabEntry) bool {
+		if entry.State == netstat.Listen && entry.LocalAddr.Port == uint16(port) {
+			return true
+		}
+		return false
+	})
+	if err != nil {
+		return -1, fmt.Errorf("failed to get pid, err is %v", err)
+	}
+	if len(socks) == 0 || socks[0].Process == nil {
+		return -1, fmt.Errorf("failed to get pid, the port is not used")
+	}
+	pid := socks[0].Process.Pid
+	portToPidMap.Store(port, pid)
+	go func() {
```

Comment:
each call of PortToPid will create a new goroutine, consider define a lastUpdateTime before do the netstat